File: 8kyu - Double Char.js
// 8kyu - Double Char

// Given a string, you have to return a string in which each character (case-sensitive) is repeated once

// const doubleChar = str => [...str].map((c => c + c)).join``
const doubleChar = str => [...str].map((char => char.repeat(2))).join``

// const doubleChar = str => str.replace(/(.)/g, '$1$1')
// const doubleChar = str => str.replace(/./g, '$&$&')
// const doubleChar = str => str.replace(/./gi, (l) => l + l);

// function doubleChar(str) {
//     let newStr = ''
//     for (let i of str)
//         newStr += i.concat(i)
//     return newStr
// }

q = doubleChar("abcd") // "aabbccdd"
q
q = doubleChar("Adidas") //  "AAddiiddaass"
q
q = doubleChar("1337") // "11333377"
q
q = doubleChar("illuminati") // "iilllluummiinnaattii"
q
q = doubleChar("123456") // "112233445566"
q
q = doubleChar("%^&*(") // "%%^^&&**(("
q

File: 8kyu/8kyu - Double Char.py
# 8kyu - Double Char

""" Given a string, you have to return a string in which each character (case-sensitive) is repeated once.

double_char("String") ==> "SSttrriinngg"

double_char("Hello World") ==> "HHeelllloo  WWoorrlldd"

double_char("1234!_ ") ==> "11223344!!__  "       """


# def double_char(s):
#     return ''.join([x + x for x in list(s)])

def double_char(s):
    return ''.join(c * 2 for c in s)


q = double_char("String")  # "SSttrriinngg"
q
q = double_char("Hello World")  # "HHeelllloo  WWoorrlldd"
q
q = double_char("1234!_ ")  # "11223344!!__  "
q
